### DISCUSSION TITLE
Deployed NextJS app's API returns 404

### DISCUSSION THREAD

**Discussion by nikolagigic** [vercel/vercel#8774] [Help]
Hey everyone,
So I've been struggling with this for past day and can't figure out why when I try to make a POST request to my API endpoint it keeps on returning 404 when the application is deployed on Vercel - it works perfectly fine locally.
I went through all [checks](https://vercel.com/guides/why-does-my-serverless-function-work-locally-but-not-when-deployed) and sorted them out if there were any on why my Serverless Function might not work and I still get the same issue.

Any idea why else could this happen?

Front-end is sending a `Blob` through `FormData` to back-end which then uses `formidable` to parse the file and perform side-effects with it. App is in **Turborepo**.

**Comment by amyegan** [comment]
Hi @nikolagigic! There are many possible causes so it's hard to say what the problem is without digging into the app. I do you see any more details in your [function logs](https://vercel.com/guides/where-are-vercel-logs#function-logs)?

If you are able to share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), I would be happy to take a closer look to help with debugging

**Reply by chanmyaemaung** [reply]
https://stackoverflow.com/questions/74250442/deployed-nextjs-apps-api-returns-404-with-nextjs-version-13

The issues is the same with him.

**Comment by nikolagigic** [comment]
@amyegan Thanks for getting back to me on this!
Is it possible that Serverless Function returns 404 because of memory usage?

Here are screenshots from Functions of the deployed app.
Some context behind it:

- `testApi` takes a file in request and does nothing with it
- `create-mux-asset` takes a file in request and process it upon receiving it

<img width="1312" alt="Screenshot 2022-10-25 at 22 00 26" src="https://user-images.githubusercontent.com/26679866/197883197-7144a154-b09e-48b9-ab83-5ebee080a6a4.png">
<img width="1357" alt="Screenshot 2022-10-25 at 21 57 50" src="https://user-images.githubusercontent.com/26679866/197883203-3b77f609-58e9-42b2-ad79-ed5cbeaf0a7a.png">

**Reply by behrangsa** [reply]
Try adding a `console.log('Start...')` at the top of your function to see if the request reaches the function at all.

**Reply by amyegan** [reply]
Behrang has the right idea. I like to add some log statements to see what data is moving through the Serverless Function and where it fails.
